lib/httparty/request.rb

```diff
@@ -276,11 +276,10 @@ def get_charset
     end
 
     def encode_with_ruby_encoding(body, charset)
-      if !body.nil? && Encoding.name_list.include?(charset)
-        body.force_encoding(charset)
-      else
-        body
-      end
+      encoding = Encoding.find(charset)
+      body.force_encoding(charset)
+    rescue ArgumentError
```

Comment:
Oof, nevermind entirely! There’s already a test that does this and I missed it: https://github.com/jnunemaker/httparty/blob/master/spec/httparty/request_spec.rb#L488-L493

So… it seems like the `nil?` check is entirely unnecessary here *and* it’s well-covered by tests. (I’m guessing it became unnecessary sometime after c46663d73ff96d86b92d36208a021bbe58672e7c)